modules/azurerm_vm/main.tf

```diff
@@ -83,11 +83,20 @@ resource "azurerm_linux_virtual_machine" "vm" {
 
   source_image_reference {
     publisher = var.os_publisher
-    offer     = var.os_offer
-    sku       = var.os_sku
+    offer     = var.fips_enabled ? "0001-com-ubuntu-pro-focal-fips" : var.os_offer
+    sku       = var.fips_enabled ? "pro-fips-20_04-gen2" : var.os_sku
```

Comment:
Does this work each time? Wondering as the Pro versions of Ubuntu usually have some cost or subscription. Guessing SAS may have this but will others? I am guessing they'll need that with their Azure subscription. Just wanting to be sure this isn't working behind the scenes and would break in the field.

Reply:
This Ubuntu FIPS image is the first and only FIPS certified image for the public cloud. This is available in Azure Marketplace and should be available for everyone to use, see: https://azuremarketplace.microsoft.com/en-us/marketplace/apps/canonical.0001-com-ubuntu-pro-focal-fips?tab=overview

This does come with some added cost, see pricing here: https://azuremarketplace.microsoft.com/en-us/marketplace/apps/canonical.0001-com-ubuntu-pro-focal-fips?tab=PlansAndPrice

To use this image in your subscription you need to accept terms before deploying cluster. To do so you need run the following az command:
`az vm image terms accept --urn Canonical:0001-com-ubuntu-pro-focal-fips:pro-fips-20_04-gen2:latest --subscription $subscription_id`
See my comment related to this in the code: https://github.com/sassoftware/viya4-iac-azure/blob/IAC-886/variables.tf#L56-L57

Reply:
FIPS support itself is not documented yet, this is being introduced as an Experimental feature. Not sure how to proceed with documentation on this specific feature. SAS Viya Platform deployments are not supported on FIPS enabled clusters yet. We will probably be announcing/documenting this feature with same timeline as SAS Viya Platform.

Reply:
Got it. Marking as resolved.